sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py

```diff
@@ -110,7 +110,7 @@ def __init__(
         compute: Optional[str] = None,
         inputs: Optional[Dict] = None,
         outputs: Optional[Dict[str, Union[str, Output, "Output"]]] = None,
-        retry_settings: Optional[Union[RetrySettings, Dict]] = None,
+        retry_settings: Optional[Union[RetrySettings, Dict[str, str]]] = None,
         logging_level: Optional[str] = None,
         max_concurrency_per_instance: Optional[int] = None,
         error_threshold: Optional[int] = None,
```

Comment:
Below on line 119, we changed `task`'s type annotation from `task: Optional[Dict[str, Union[ParallelTask, str]]]`. Why is it `Optional[Union[ParallelTask, RunFunction, Dict]]` now?

Reply:
the original task type was wrong if it was `task: Optional[Dict[str, Union[ParallelTask, str]]] = None`

![image](https://github.com/Azure/azure-sdk-for-python/assets/19650434/cdccbb0c-c047-41fb-b174-9b09344a83c3)